graylog2-server/src/main/java/org/graylog/plugins/views/search/validation/validators/FieldValueTypeValidator.java

```diff
@@ -0,0 +1,146 @@
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.views.search.validation.validators;
+
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.graylog.plugins.views.search.ParameterProvider;
+import org.graylog.plugins.views.search.Query;
+import org.graylog.plugins.views.search.elasticsearch.QueryParam;
+import org.graylog.plugins.views.search.elasticsearch.QueryStringDecorators;
+import org.graylog.plugins.views.search.engine.PositionTrackingQuery;
+import org.graylog.plugins.views.search.engine.QueryPosition;
+import org.graylog.plugins.views.search.errors.EmptyParameterError;
+import org.graylog.plugins.views.search.errors.MissingEnterpriseLicenseException;
+import org.graylog.plugins.views.search.errors.SearchException;
+import org.graylog.plugins.views.search.errors.UnboundParameterError;
+import org.graylog.plugins.views.search.rest.MappedFieldTypeDTO;
+import org.graylog.plugins.views.search.validation.FieldTypeValidation;
+import org.graylog.plugins.views.search.validation.LuceneQueryParser;
+import org.graylog.plugins.views.search.validation.ParsedQuery;
+import org.graylog.plugins.views.search.validation.QueryValidator;
+import org.graylog.plugins.views.search.validation.ValidationContext;
+import org.graylog.plugins.views.search.validation.ValidationMessage;
+import org.graylog.plugins.views.search.validation.ValidationRequest;
+import org.graylog.plugins.views.search.validation.ValidationStatus;
+import org.graylog.plugins.views.search.validation.ValidationType;
+import org.graylog2.indexer.fieldtypes.FieldTypes;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class FieldValueTypeValidator implements QueryValidator {
+
+    private final FieldTypeValidation fieldTypeValidation;
+    private final LuceneQueryParser luceneQueryParser;
+    private final QueryStringDecorators queryStringDecorators;
+
+    @Inject
+    public FieldValueTypeValidator(FieldTypeValidation fieldTypeValidation, LuceneQueryParser luceneQueryParser, QueryStringDecorators queryStringDecorators) {
+        this.fieldTypeValidation = fieldTypeValidation;
+        this.luceneQueryParser = luceneQueryParser;
+        this.queryStringDecorators = queryStringDecorators;
+    }
+
+    @Override
+    public List<ValidationMessage> validate(ValidationContext context) {
+        try {
+            PositionTrackingQuery decorated = decoratedQuery(context.request());
+            return validateQueryValues(decorated, context.availableFields());
+        } catch (ParseException e) {
+            return Collections.singletonList(ValidationMessage.fromException(e));
+        } catch (SearchException searchException) {
+            return convert(searchException);
```

Comment:
I'd consider calling this method more explicitly (convertToValidationMessage ? buildValidationMessage ?).

Reply:
Moved and unified, thx!